contracts/ERC20.sol

```diff
@@ -0,0 +1,29 @@
+//SPDX-License-Identifier: mit
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/ERC20.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract ExampleERC20 is ERC20, AccessControl {
+  string private TOKEN_NAME = "Example ERC20 Token";
+  string private TOKEN_SYMBOL = "XMPL";
+
+  uint private constant TOTAL_SUPPLY = 123456789;
+
+  constructor() ERC20(TOKEN_NAME, TOKEN_SYMBOL) {
+    _mint(msg.sender, TOTAL_SUPPLY);
+    _setupRole(DEFAULT_ADMIN_ROLE, msg.sender);
+  }
+
+  function decimals() public view virtual override returns (uint8) {
```

Comment:
Maybe let's push people to make 18 decimal tokens. We could do the override or just delete this method.

Reply:
Resolved